A36-xds-for-servers.md

```diff
@@ -192,8 +190,48 @@ required. This is to allow implementations that only support L4 xDS features to
 avoid L7 plumbing and implementation. This has no impact on the resource
 validation and NACKing behavior called for in A39.
 
+If an XdsServer implementation uses RouteConfiguration or supports any HTTP
+filters other than the hard-coded Router, then
+`HttpConnectionManager.route_config` and `HttpConnectionManager.rds` must be
+supported and RouteConfigurations must be validated. RouteConfiguration
+validation logic inherits all previous validations made for client-side usage
+as RDS does not distinguish between client-side and server-side. That is
+predomenently defined in [gRFC A28][A28-validation], although note that
+configuration for all VirtualHosts have been validated on client-side since
+sharing the XdsClient was introduced, yet was not documented in a gRFC. The
+validation must be updated to allow [Route.non_forwarding_action][] as a valid
+`action`. The VirtualHost is selected on a per-RPC basis using the RPC's
+requested `:authority`. Routes are matched the same as on client-side.
+`Route.non_forwarding_action` is expected for all Routes used on server-side and
+`Route.route` continues to be expected for all Routes used on client-side; a
+Route with an inappropriate `action` causes RPCs matching that route to fail
+with UNAVAILABLE. If `HttpConnectionManager.rds` references a NACKed resource
+without a previous good version, an unavailable resource because of
+communication failures with control plane or a triggered loading timeout, or
+a non-existent resource, then all RPCs processed by that HttpConnectionManager
+will fail with UNAVAILABLE.
+
+[Like in Envoy][envoy lds], updates to a Listener cause all older connections on
+that Listener to be gracefully shut down (i.e., "drained") with a default grace
+period of 10 minutes for long-lived RPCs, such that clients will reconnect and
+have the updated configuration apply. This applies equally to an update in a
+RouteConfiguration provided inline via the `route_config` field as it is part of
+the Listener, but it does not apply to an updated RouteConfiguration provided by
+reference via `rds` field. Infrastructure for "not serving" mode may be used for
```

Comment:
I don't understand what this sentance means:

"""
Infrastructure for "not serving" mode may be used for the graceful shutdown, but only if it can be used without causing spurious RPC or connection failures.
"""

If we get an LDS update, then presumably we should switch over to the new config for incoming connections ~immediately (modulo a short delay to get an RDS resource, if any, but that will be shorter than the 10m drain grace time), so there should be no period of time during which we are not listening to report as "not serving".